hack/update-codegen.sh

```diff
@@ -114,3 +114,13 @@ informer-gen \
   --versioned-clientset-package=github.com/karmada-io/karmada/pkg/generated/clientset/versioned \
   --listers-package=github.com/karmada-io/karmada/pkg/generated/listers \
   --output-package=github.com/karmada-io/karmada/pkg/generated/informers
+
+echo "Generating with openapi-gen"
+GO111MODULE=on go install k8s.io/code-generator/cmd/openapi-gen
+openapi-gen \
+  --go-header-file hack/boilerplate/boilerplate.go.txt \
+  --input-dirs "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1" \
+  --input-dirs "k8s.io/api/core/v1,k8s.io/apimachinery/pkg/api/resource" \
+  --input-dirs "k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/runtime,k8s.io/apimachinery/pkg/version" \
```

Comment:
Is it directly written as fixed here? I've seen a recursive way of generating.

Reply:
@XiShanYongYe-Chang 

We must specify the dependencies as fixed here. without it, the karmada-aggregated-apiserver will fail to start.

```shell
➜  ~ kubectl -n karmada-system logs karmada-aggregated-apiserver-b8876cfdb-s4pgt
I0104 05:41:41.811682       1 plugins.go:158] Loaded 2 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,MutatingAdmissionWebhook.
I0104 05:41:41.811800       1 plugins.go:161] Loaded 1 validating admission controller(s) successfully in the following order: ValidatingAdmissionWebhook.
Error: unable to get openapi models: cannot find model definition for k8s.io/apimachinery/pkg/api/resource.Quantity. If you added a new type, you may need to add +k8s:openapi-gen=true to the package or type and run code-gen again
Usage:
  karmada-aggregated-apiserver [flags]
  karmada-aggregated-apiserver [command]

...
```

Reply:
code generator provides a script `generate-internal-groups.sh` to generate internal groups, can we use it?

https://github.com/kubernetes/code-generator/blob/05a01c0d31e8d2d8ecc63342181c8fe302d1c04c/generate-internal-groups.sh#L21-L22

Reply:
@XiShanYongYe-Chang we can use it, but it still requires that we specify the missing dependencies. for example, `k8s.io/apimachinery/pkg/api/resource`

Reply:
Where does `openapi-gen` gets the `--input-dirs`? From vendor or `GOPATH/src/`?

I'm thinking if it still works if the dependency packages do not exit in `GOPATH/src/`, that might happen in our CI system.

Reply:
I test it on my local, it gets from the vendor.